### PR TITLE
Switch exec to look at exit code not output status.

### PR DESCRIPTION
**Pull request by brendandburns**
I have run e2e and they pass.  Since this is a breaking change, let's wait until after the 0.17.0 cut to merge, so that people have 2 weeks to change anything they're currently using.

### CODE REVIEW

**Comment by zmerlynn** [comment]
Assigning, but @vishh, note the ask not to merge.

File: pkg/kubelet/dockertools/manager.go [hunk 1]
```diff
@@ -894,10 +895,47 @@ func (dm *DockerManager) RunInContainer(containerID string, cmd []string) ([]byt
 		RawTerminal:  false,
 	}
 	err = dm.client.StartExec(execObj.ID, startOpts)
+	if err != nil {
+		return nil, err
+	}
+	tick := time.Tick(2 * time.Second)
+	for {
```
Comment (vishh) [inline pkg/kubelet/dockertools/manager.go]: Should we have some timeout here, just to be safe?

**Comment by vishh** [comment]
LGTM excepting the one comment.

File: pkg/kubelet/dockertools/manager.go (hunk 1 above)
Comment (brendandburns) [inline pkg/kubelet/dockertools/manager.go]: Hrm, we were already blocking (detach is set to 'false') so to be honest this loop is probably never actually executed.

I'd rather not have a timeout here.

**Comment by brendandburns** [comment]
Commented addressed (well at least commented on), please re-check.

Thanks
--brendan

File: pkg/kubelet/dockertools/manager.go (hunk 1 above)
Comment (vishh) [inline pkg/kubelet/dockertools/manager.go]: Ack

**Pull request merged**